pageserver/src/tenant/storage_layer.rs

```diff
@@ -188,7 +292,7 @@ pub trait PersistentLayer: Layer {
     /// current_physical_size is computed as the som of this value.
     fn file_size(&self) -> Option<u64>;
 
-    fn info(&self) -> HistoricLayerInfo;
+    fn info(&self, reset: Option<LayerAccessStatsReset>) -> HistoricLayerInfo;
```

Comment:
Some comment on how that's used would be good to have.

Reply:
For clarity, I'd suggest splitting this into two functions:

```
/// read access stats
fn access_stats(&self) -> HistoricLayerInfo

/// Atomically read the current access stats, and reset them 
fn reset_access_stats(&self, opts: LayerAccessStatsReset) -> HistoricLayerInfo
```

Reply:
> For clarity, I'd suggest splitting this into two functions:

Was hesitant because it's another method on `PersistentLayer` that needs to be implemented everywhere.

How about instead of `Option<LayerAccessStatsReset>`, it's just `LayerAccessStatsReset::{NoReset,JustTaskKindFlags,AllStats}`. 

That way, the call site is self-documenting. Instead of
```
.access_stats(None)
```
we'd have
```
.access_stats(LayerAccessStatsReset::NoReset)
```

Reply:
Yeah the 3-state is fine, agreed on less assorted trait methods the better.

Reply:
Pushed the 3-sate